### PR TITLE
DR-1013 Implement preprod worflow

### PR DESCRIPTION
**Pull request by Pira-nhs**
## Description

Re-usuable workflow to derive artefact address

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by gdcNHS — APPROVED** [review]

File: .github/actions/derive-artefact-address/action.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: "Collect artefact address action"
+description: "Collects the artefact sub directory and commit has for pipeline being executed"
+
+inputs:
+  tag:
+      description: "The git tag to derive artefact location from"
+      required: true
+outputs:
+  commit_hash:
+    description: "Commit hash for head of branch being processed"
```
Comment (timrickwood) [inline .github/actions/derive-artefact-address/action.yaml:10]: Feel free to ignore but i think this is the commit hash relating to the git tag - it is not necessarily the hash at the head of the branch

File: .github/actions/derive-artefact-address/action.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    value: ${{ steps.artefact-address.outputs.commit_hash }}
+  artefact_sub_dir:
+    description: "The unique commit hash for artefacts for flows triggered by a pull request "
```
Comment (timrickwood) [inline .github/actions/derive-artefact-address/action.yaml:13]: Is this right ? Is this better described as the parent directory for the built artefacts... or something similar

File: .github/workflows/derive-artefact-address.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+name: Derive artefact address workflow
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: "The git tag to derive artefact location from"
+        required: true
+        type: string
+      workflow_timeout:
```
Comment (timrickwood) [inline .github/workflows/derive-artefact-address.yaml:10]: is there a reason why we opted to pass this. I know for some jobs (eg spinning up cloudfront etc) we wanted to be able to extend the github default but this one looks pretty self-contained. Happy to keep it if we need it but if we don;t maybe its cleaner if we don't specify it as an optional input

File: .github/actions/derive-artefact-address/action.yaml (hunk 1 above)
Comment (Pira-nhs) [reply .github/actions/derive-artefact-address/action.yaml:10]: have updated description to suit the use better

File: .github/actions/derive-artefact-address/action.yaml (hunk 2 above)
Comment (Pira-nhs) [reply .github/actions/derive-artefact-address/action.yaml:13]: have updated description to suit the use better

File: .github/workflows/derive-artefact-address.yaml (hunk 3 above)
Comment (Pira-nhs) [reply .github/workflows/derive-artefact-address.yaml:10]: Removed timeout

**Review by gdcNHS — APPROVED** [review]

**Review by timrickwood — APPROVED** [review]

**Pull request merged**